lens-labels/src/Lens/Labels.hs

```diff
@@ -82,13 +82,26 @@ instance
 #endif
 
 -- | A type class for lens fields.
-class HasLens' s (x :: Symbol) a | s x -> a where
+--
+-- Note: in order to support better type error messages, this class does not have
+-- a fundep relationship to the parameter @a@.  Instead, instances are expected to
+-- be defined such that @a@ is derivable from @s@ and @x@.  For example, to define a
+-- field named @"x"@ of type @Int32@ for the containing type @Foo@:
+--
+-- > instance a ~ Int32 => HasLens' Foo "x" a where ...
```

Comment:
Could you comment on why this is preferrable
`instance a ~ Int32 => HasLens' Foo "x" a where ...`
to this:
`instance HasLens' Foo "x" Int32 where ...`

Reply:
I think I recognized the constraint trick, still makes sense to document it as such.

Reply:
Added more explanation.